### PR TITLE
WIP support for multiple paths for Hadoop dist cache #144

### PR DESCRIPTION
**Pull request by andrewsmartin**
@nevillelyh @ravwojdyla RFC - would like to get your input on this. Since Scala doesn't allow overridden methods with default parameters, I saw 2 options:

1) remove the default parameters and instead have a slew of overloaded methods, or
2) use a builder. This is currently what is in place but unfortunately this would be a breaking change.

Thoughts?

### CODE REVIEW

**Comment by nevillelyh** [comment]
Sorry just got a chance to look at this.
How about just use a different name this time, e.g. `def hadoopDistCacheMulti` and maintain argument consistency with the core version?

**Comment by andrewsmartin** [comment]
@nevillelyh Updated with your suggestion

File: scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala [hunk 1]
```diff
@@ -121,41 +121,58 @@ package object hdfs {
...
+    /**
+     * Create a new [[com.spotify.scio.values.DistCache DistCache]] instance for a list of
+     * files on Hadoop/HDFS.
+     *
+     * @param paths Sequence of paths to the Hadoop/HDFS artifacts
+     * @param conf optional custom Hadoop configuration
+     * @param username optional Hadoop Simple Authentication remote username
+     */
+    def hadoopDistCacheMulti[F](paths: Seq[String],
+                                   conf: Configuration = null,
```
Comment (nevillelyh) [inline scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala]: fix indentation?

File: scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala [hunk 2]
```diff
@@ -121,41 +121,58 @@ package object hdfs {
...
+        require(!paths.contains(null), "Artifact path can't be null")
 
         val _conf = Option(conf).getOrElse(new Configuration())
 
-        //TODO: should we add checksums on both src and GCS to reuse uploaded artifacts?
-        // to keep it simple, for now we upload each artifact, thus artifacts should be
-        // relatively small.
-        val pathHash = Hashing.sha1().hashString(path, Charsets.UTF_8)
-        val targetHash = pathHash.toString.substring(0, 8)
+        val targetPaths = paths.foldLeft(Seq[Path]()) { (targets, path) =>
```
Comment (nevillelyh) [inline scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala]: looks like this can be done with a `paths.map`?

Comment (andrewsmartin) [inline scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala]: Woops. Not entirely sure what I was thinking

**Comment by nevillelyh** [comment]
LGTM, merge when green.

**Pull request merged**